### DISCUSSION TITLE
.net 5 File.Create create file incorrect directory , path

### DISCUSSION THREAD

**Discussion by uopeydel** [dotnet/dotnet-docker#2565] [Q&A] [answered]
### Describe the Bug

I try to create file in docker container. It can create. But create incorrect directory. When create in local(windows) it correct.

### Steps to Reproduce

My code to create file is 
```
                        var dir = Directory.GetCurrentDirectory() + "\\wwwroot\\xls_template\\";
                        if (!Directory.Exists(dir))
                        {
                            Directory.CreateDirectory(dir); 
                        }
                        dir = dir + $"xlssave_{DateTime.UtcNow.AddHours(7).ToString("dd_MM_yyyy___HH_mm")}.xlsx"; 
                        using (var fileStream = File.Create(dir))
                        {
                            stream.Seek(0, SeekOrigin.Begin);
                            stream.CopyTo(fileStream);
                        }

```
My Dockerfile just this .. ** I build+publish by command to target folder instead build in dockerfile. It build correct and can surf all feature that I have correct.

```

FROM mcr.microsoft.com/dotnet/aspnet:5.0.2-focal-amd64 AS runtime
RUN apt-get update \
    && apt-get install -y --allow-unauthenticated \
        libc6-dev \
        libgdiplus \
        libx11-dev \
     && rm -rf /var/lib/apt/lists/*

WORKDIR /app
COPY PublishFile/* /app/
RUN mkdir wwwroot
RUN cp -avr xls_template/ wwwroot/
RUN ls
RUN pwd
ENTRYPOINT ["dotnet", "SMS.Server.dll","--environment=UAT"]
```

This image below is happen when I use command exec into docker container is running.
It create my file at '/'

-rw-r--r--.   1 root root 7.2K Jan 31 09:30 'app\wwwroot\xls_template\xlssave_31_01_2021___16_30.xlsx'
-rw-r--r--.   1 root root 7.2K Jan 31 09:31 'app\wwwroot\xls_template\xlssave_31_01_2021___16_31.xlsx'
-rw-r--r--.   1 root root 7.2K Jan 31 09:32 'app\wwwroot\xls_template\xlssave_31_01_2021___16_32.xlsx'
-rw-r--r--.   1 root root 7.2K Jan 31 09:33 'app\wwwroot\xls_template\xlssave_31_01_2021___16_33.xlsx'

this is my file is created.

![image](https://user-images.githubusercontent.com/12793024/106380990-70257e80-63e8-11eb-8425-b665244a636e.png)
 
### Output of `docker version`

 mcr.microsoft.com/dotnet/aspnet:5.0.2-focal-amd64

**Comment by MichaelSimons** [comment, marked as answer]
It looks like this is an issue related to Windows and Linux using different path separators - `\` vs `/`.  Linux does not detect `\` as a valid path separator and therefore treats is as a character in the directory and file names.  To make the app cross platform compatible, consider using [Path.PathSeparator](https://docs.microsoft.com/en-us/dotnet/api/system.io.path.pathseparator?view=net-5.0) or [Path.Combine](https://docs.microsoft.com/en-us/dotnet/api/system.io.path.combine?view=net-5.0) to build up the paths.
